src/mpid/ch3/src/ch3u_request.c

```diff
@@ -98,8 +98,8 @@ int MPIDI_CH3U_Request_load_send_iov(MPIR_Request * const sreq,
     int max_iov_len = *iov_n;
     MPI_Aint actual_iov_bytes;
     MPIR_Typerep_to_iov(sreq->dev.user_buf, sreq->dev.user_count, sreq->dev.datatype,
-                     sreq->dev.msg_offset, iov, max_iov_len,
-                     sreq->dev.msgsize - sreq->dev.msg_offset, iov_n, &actual_iov_bytes);
+                     sreq->dev.msg_offset, iov, (MPI_Aint) max_iov_len,
+                     sreq->dev.msgsize - sreq->dev.msg_offset, (MPI_Aint *) iov_n, &actual_iov_bytes);
```

Comment:
Pointer casting this way is not safe.  Why is `iov_n` not upgraded to `MPI_Aint` instead?

Reply:
`int * const iov_n` is passed in from upper level.

Reply:
In that case you'd need to use a different variable and cast it after the function returns.

Reply:
Instead, I fixed all the places in `ch3` to use `dev.iov_count` as `MPI_Aint`. Then there is no need for using an extra varaible.

Reply:
There is a `tcp` failure after set `dev.iov_count` to be `MPI_Aint`, I switch it back to use an extra `MPI_Aint` variable for now.